### PR TITLE
[full-ci] Run e2e tests in ocis CI

### PR DESCRIPTION
**Pull request by ScharfViktor**
- [x] run e2e tests in ocis CI

### CODE REVIEW

**Comment by fschade** [comment]
@ScharfViktor, thx buddy for taking care ❤️ , i have some questions before merge:

* can we somehow see / test if publishing of e2e test traces is working here (uploadTracingResult & publishTracingResult)?
* drone is red atm, is this intended?
* before we merge, we need to bring back skipped tests and have one green test-run.

**Comment by phil-davis** [comment]
"This branch is [2 commits ahead](https://github.com/owncloud/ocis/compare/master...runE2eTestsInCi), [366 commits behind](https://github.com/owncloud/ocis/compare/runE2eTestsInCi...master) master."

And need to rebase to make sure all is good.

**Comment by ScharfViktor** [comment]
> "This branch is [2 commits ahead](https://github.com/owncloud/ocis/compare/master...runE2eTestsInCi), [366 commits behind](https://github.com/owncloud/ocis/compare/runE2eTestsInCi...master) master."
> 
> And need to rebase to make sure all is good.

I'll do it

**Comment by ScharfViktor** [comment]
> 

I think it's important to have tracing here. I need to finish it. How do you think, do we need to publish the result on github or is it enough to have the result in drone?

**Comment by ScharfViktor** [comment]
> @ScharfViktor, thx buddy for taking care ❤️ , i have some questions before merge:
> 
> * can we somehow see / test if publishing of e2e test traces is working here (uploadTracingResult & publishTracingResult)?
> * drone is red atm, is this intended?
> * before we merge, we need to bring back skipped tests and have one green test-run.

I think it's important to have tracing here. I need to finish it. How do you think, do we need to publish the result on github or is it enough to have the result in drone?

**Comment by ScharfViktor** [comment]
The first run was successful: https://drone.owncloud.com/owncloud/ocis/13890/50/5
second was failed: https://drone.owncloud.com/owncloud/ocis/13892/50/6

In ocis CI e2e the tests are more flaky. I don't understand why this is happening. 
@fschade do we need build web `yarn build` somewhere?

**Comment by fschade** [comment]
Not sure, I have to dig a bit deeper to give a valid answer

**Comment by fschade** [comment]
I would preferred to have it as gh comment, but as a first step drone is enough (my 5ct)

**Comment by ScharfViktor** [comment]
> Not sure, I have to dig a bit deeper to give a valid answer

The test always falls in one place:
https://drone.owncloud.com/owncloud/ocis/13906/50/6
trace for that: `npx playwright show-trace https://cache.owncloud.com/public//13906/tracing/folder-brian-2022-7-28-03-18-56.zip`

**Comment by ScharfViktor** [comment]
We don't do `yarn build:w` and we have a old web with new code

In this commit https://github.com/owncloud/web/commit/7747d8385ef7ddc81a86e97ae233ee946190973b#diff-bdf82996a7fca0461152dfa6620089e336592e6a02b7cb5854e1d3fab3c1cd3b changed:
```
- const filesSharedWithMe = `#files-shared-with-me-shares-table [data-test-resource-name="%s"]`
+ const filesSharedWithMeAccepted = `#files-shared-with-me-accepted-section [data-test-resource-name="%s"]`

and test is looking in the old new element: `#files-shared-with-me-accepted-section [data-test-resource-name="folder_to_shared"]"` but it's not exist. 
```

Screenshot 2022-07-29 at 14 20 26

**Comment by ScharfViktor** [comment]
@fschade thank you for your hint  ❤️  
e2e tests are green. https://drone.owncloud.com/owncloud/ocis/13933/76/5

- tests use the latest stable version of the web (as a ocis)
- tests run if: have tag in title [full-ci], have event "cron"

**Comment by ScharfViktor** [comment]
https://drone.owncloud.com/owncloud/ocis/13950/76/5
Look like laky test witch was fixed here: https://github.com/owncloud/web/pull/7278

**Comment by fschade** [comment]
@ScharfViktor you say

>  tests run if: have tag in title [full-ci], have event "cron"

can you explain what you mean with `have event "cron"`

**Comment by ScharfViktor** [comment]
> @ScharfViktor you say
> 
> > tests run if: have tag in title [full-ci], have event "cron"
> 
> can you explain what you mean with `have event "cron"`

running tests at night

File: .drone.env
Comment (individual-it) [inline .drone.env:10]: shouldn't that be the same commit at `WEB_COMMITID`? why do we need that?

Comment (fschade) [reply .drone.env:10]: it is used here to download the web version -> https://github.com/owncloud/ocis/blob/26f15c2fa9abd1767eb0f7d60200404a613a7dc3/services/web/Makefile#L36

Comment (fschade) [reply .drone.env:10]: WEB_COMMITID is only used at `settingsUITests ` & `yarnInstallUITests`

**Review by fschade — APPROVED** [review]
LGTM 🤖 , please wait for the ok from @individual-it before merging

File: .drone.env
Comment (ScharfViktor) [reply .drone.env]:
```
"git clone -b $WEB_ASSETS_VERSION --single-branch --no-tags https://github.com/owncloud/web.git folder"
```
is same 
```
"git clone -b $WEB_BRANCH --single-branch --no-tags https://github.com/owncloud/web.git folder"
"git checkout $WEB_COMMITID",
```

if WEB_COMMITID matches with WEB_ASSETS_VERSION of web.  I tried to run it again in CI twice and it looks like it works

**Review by individual-it — APPROVED** [review]

**Comment by individual-it** [comment]
we could add also some skip-if-unchanged, but that can be done in an other PR

**Pull request merged**
